### DISCUSSION TITLE
How to use moose-based app2 in moose-based app1?

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27460] [Q&A General] [answered]
Hi all,

I  have developed two moose-based applications. Now, I would like to use the source code app2 (also run cases of app2) in app1, just like using modules of moose in app1. How to modify the Makefile of app1? Both app1 and app2 are in projects dir.
I tried to modify the makefile of app1, but it doesn't work.
[Makefile.txt](https://github.com/idaholab/moose/files/15094736/Makefile.txt)

Could anyone kindly help me? Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

There is an example of building two moose-based applications together in Cardinal, which can be built with many other apps
https://github.com/neams-th-coe/cardinal/blob/devel/Makefile
It does come by modifying the makefile, but not just that. Also the src/base/...App.C file

Another option is to avoid building them together, but use dynamic loading instead
https://mooseframework.inl.gov/source/actions/DynamicObjectRegistrationAction.html

Guillaume

**Reply by SomnusYu** [reply]
Thanks!
